src/TSTransformer/nodes/transformOptionalChain.ts

```diff
@@ -257,10 +259,12 @@ function transformOptionalChainInner(
 			if (item.callOptional) {
 				if (item.kind === OptionalChainItemKind.PropertyCall) {
 					baseExpression = luau.property(convertToIndexableExpression(baseExpression), item.name);
-				} else {
+				} else if (item.kind === OptionalChainItemKind.ElementCall) {
```

Comment:
This kind is already the only remaining one in this statement - explicitly if-ing on it leaves an edge case if the kind would ever be expanded. I also don't see the need to change it for this PR specifically.